front/src/utils/timeManipulation.ts

```diff
@@ -1,18 +1,27 @@
 import * as d3 from 'd3';
 import { TimeString } from 'common/types';
 
-export function datetime2string(ts: string | number | Date) {
+export function datetime2string(ts: string | number | Date): TimeString {
   const datetime = new Date(ts);
   return datetime.toLocaleString();
 }
 
+export function datetime2Isostring(ts: string | number | Date): TimeString {
+  const datetime = new Date(ts);
+  return datetime.toISOString();
+}
+
 export function datetime2time(datetime: Date) {
   const formatTime = d3.timeFormat('%H:%M:%S');
   return formatTime(datetime);
 }
 
 export function time2datetime(time: TimeString) {
-  return d3.timeParse('%H:%M:%S')(time);
+  return d3.timeParse('%H:%M:%S')(time as string);
```

Comment:
Why do you need to cast here, as TimeString is a string ?
```suggestion
  return d3.timeParse('%H:%M:%S')(time);
```

Reply:
yes